packages/astro/src/content/runtime.ts

```diff
@@ -107,6 +104,8 @@ export function createGetCollection({
 			);
 			cacheEntriesByCollection.set(collection, entries);
 		}
+		// Always return a new instance so consumers can safely mutate it
+		entries = [...cacheEntriesByCollection.get(collection)!];
```

Comment:
This can be expensive for large collections. Given this is corner case, can we use `Object.freeze` when we set it instead? That would have similar effect (can't push, splice, etc) but without doing it on ever request.

Reply:
- Slice method https://measurethat.net/Benchmarks/Show/2667/0/arrayprototypeslice-vs-spread-operator#latest_results_block seems to be much faster.
- Can not freeze the cached collection as individual pages can still sort the collection based on it's attributes as done in https://github.com/withastro/astro/packages/astro/test/content-collections-render.test.js fixture: https://github.com/withastro/astro/blob/main/packages/astro/test/fixtures/content/src/pages/sort-blog-collection.astro#L7.

Reply:
> Can not freeze the cached collection as individual pages can still expect to sort the collection

Where did you read this? I just tried to use `Object.freeze`, and it works